### DISCUSSION TITLE
DeDRM Compatible with Calibre 6

### DISCUSSION THREAD

**Discussion by JSWolf** [noDRM/DeDRM_tools#54] [Q&A]
Is DeDRM compatible with Calibre 6 or will it be by the time Calibre 6 is released?

**Comment by noDRM** [comment]
The latest release (v10.0.2) is not, but the current in-progress version on the master branch is, and that'll probably be released as v10.0.3 relatively soon.

**Reply by birgerjrj** [reply]
Still "relatively soon"?

**Reply by JasonUK67** [reply]
Calibre v6.0 now out.  Any news on DeDRM v10.0.3?

**Reply by sloping0083** [reply]
of course the day I need to remove DRM a new version of calibre came out which I didn't notice until I updated it already :)

**Reply by jwilker** [reply]
Same! Came looking for details and womp womp. Now I gotta make sure to keep the file around so I can load it when this is updated LOL

**Reply by noDRM** [reply]
I guess the latest master has already been used by so many people that I could just release it without much more testing. 
A new Calibre-6-related issue has come up ( #96 ) in the Kobo plugin, I'm going to wait a couple days to see if that bugfix works and then there'll be 10.0.3 which most likely will be nearly identical to the latest code on master.

**Reply by noDRM** [reply]
Since multiple people already confirmed that that bugfix works, I've just released 10.0.3.

**Reply by MisterRat** [reply]
Thank you

**Reply by Tahini-123** [reply]
It seems that the newly released Calibre version 6.1 blacklisted the DeDRM plugin - officially only the old version 7.2.1, but I am unable to install version 10.0.3 on OSX. It would appear that there is an issue either in the plugin not declaring correctly its version or in Calibre's blacklisting rule.

**Reply by noDRM** [reply]
I've checked the Calibre code, the blacklisting only applies to 10.0.2 and lower. 10.0.3 should install just fine, and I've seen multiple reports from MacOS users with Calibre 6.1 who said 10.0.3 works fine. 
Maybe try completely removing the current version, then restart Calibre, then try re-installing 10.0.3?

**Reply by ElleKayEm** [reply]
It looks like DeDRM 10.0.3 does have its version number correct: https://github.com/noDRM/DeDRM_tools/blob/master/DeDRM_plugin/__version.py

Are you sure you were trying to install 10.0.3?  Since calibre will open the same folder from the last install, it's easy to accidentally grab an older version -- ask me how I know!

**Reply by ZolaLa9** [reply]
FWIW, I just downloaded Calibre 6.1 to my Mac (Catalina) and installed 10.0.3 without any problems. Working fine.

**Reply by Githubmoses** [reply]
Am I missing something??  Maybe every single version back to the beginning of time has been changed by the calibre developer but today in Sept. 2022 there is no way to install external plug ins.  I tried all the way back to versions 2.x! Am I just that stupid, or am I missing something???  This is infuriating.  I actually own the physical book AND paid for the epub file and just want to look at when I want to, not on a PHONE.

**Reply by jcotton42** [reply]
@Githubmoses Preferences > Plugins > Load plugin from file

It's always been there, never gone away.

**Reply by Githubmoses** [reply]
I should have been more clear: I was attempting this on a mac.  It is most DEFINITELY not there for mac users.  You can install using command line but that's sketch and causes it to crash.  I used windows.  It installed but is useless for epub, so it's fine.

**Reply by kovidgoyal** [reply]
What nonsense. It most definitely is there on macs.

**Reply by ElleKayEm** [reply]
When you're ready for some actual help, let us know.

**Reply by Githubmoses** [reply]
I'm ready for "actual help."  You guys can mark up this image and let me know what I'm missing. 
![Screen Shot 2022-09-15 at 10 55 10 PM](https://user-images.githubusercontent.com/22313590/190553536-8eaa3d93-c935-4920-83e5-584aa5a7d16e.png)

**Reply by kovidgoyal** [reply]
Click the  Preferences item in the global menu bar, the click change calibre behavior, then click plugins, then click load plugin from file.  Be amazed.

**Reply by Githubmoses** [reply]
Now THAT, indeed, IS actual help!  Thanks!

**Comment by Tahini-123** [comment]
Confirmed working, another example of user error I guess :P

**Comment by Dunhill69** [comment]
Hi guys, thanks for this plugin.
Query can be used on linux with Calibre ?

**Reply by noDRM** [reply]
Yes, this plugin can be used on all operating systems that Calibre supports.

**Reply by Githubmoses** [reply]
It has to be installed via command line on a mac.  And I've not been successful at getting it to work.  I will attempt real linux tomorrow.

**Reply by ElleKayEm** [reply]
No command line needed, same on all versions: Preferences > Plugins > Load plugin from file

**Reply by Githubmoses** [reply]
Arrogantly repeating that doesn't make it any more true than it was the first time.  As @kovidgoyal noted above: Click the Preferences item in the global menu bar, the click "CHANGE CALIBRE BEHAVIOR", then click plugins, then click load plugin from file.

**Reply by ElleKayEm** [reply]
Works for me.  Only need the extra step if you click the drop down menu instead of the Preferences button itself.

**Comment by makjb2** [comment]
For info I think there is an "odd" circumstance where Calibre 6.1 & DeDRM 10.0.3 doesn't work on Macs.
I had a combination of an older build of DeDRM v10.0.2 working upto the last version 5 of Calibre.
Calibre v6 stopped  DeDRM working so I updated to the latest build (updated by removing my current version and adding the latest version of the plug-in). This too worked.
Updated to Calibre v6.1 and V10.0.2 was black-listed and hence DeDRM stopped working again.
Tried to update DeDRM to v10.0.3 BUT because of the blacklisting I could not uninstall the previous version of DeDRM.  This stopped the DeDRM plug-in from updating to 10.0.3.
I reported this as a bug on the Calibre site but Kovid Goyal disagreed and referred me here.
My solution - revert back to Calibre v6.0, remove DeDRM v10.0.2.  Upgraded to Calibre v6.1 then I was able to install DeDRM v10.03
 Result - everything back to normal and DeDRM working.
